### PR TITLE
Cherry-pick #7134 to 6.3: Document username/password options for HTTP based modules

### LINKED ISSUE #6493: Document http authentication for metricbeat http module
https://github.com/elastic/beats/issues/4268 is closed, but authentication for the http module is not yet documented.

This can be specified via a headers map with a base64 string (example below is for elastic:changeme) for basic authorization, eg.

```
...
...
  request.enabled: true
  response.enabled: true
  headers:
    Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==
```

### PR DESCRIPTION
**Pull request by tsg**
Cherry-pick of PR #7134 to 6.3 branch. Original message: 

This adds commented out `username` and `password` settings to most
of the config places where they are supported (this part is manual).
Since those configs are included in the docs, they also show up there.

This also adds a line in the docs that link to "HTTP specific settings"
(this part is automatised during `make update`).

Completes #4268 and closes #6493.

### CODE REVIEW

**Comment by tsg** [comment]
@carlos @jsoriano There were a couple of conflicts when backporting this. Can you give it a sanity check, please?

File: metricbeat/module/nginx/_meta/config.yml [hunk 1]
```diff
@@ -1,2 +1,8 @@
 - module: nginx
   hosts: ["http://127.0.0.1"]
+
+  # Path to server status. Default server-status
+  #server_status_path: "server-status"
```
Comment (jsoriano) [inline metricbeat/module/nginx/_meta/config.yml]: This change about the status path was not in #7134

File: metricbeat/module/php_fpm/_meta/config.yml [hunk 2]
```diff
@@ -1,2 +1,5 @@
 - module: php_fpm
   hosts: ["localhost:8080"]
+  status_path: "/status"
```
Comment (jsoriano) [inline metricbeat/module/php_fpm/_meta/config.yml]: Same here, this was not in #7134

File: metricbeat/modules.d/ceph.yml.disabled [hunk 3]
```diff
@@ -1,6 +1,9 @@
 - module: ceph
   metricsets: ["cluster_health", "cluster_status", "monitor_health"]
   period: 10s
+  hosts: ["localhost:5000"]
```
Comment (jsoriano) [inline metricbeat/modules.d/ceph.yml.disabled]: Not in #7134.

File: metricbeat/module/kubernetes/_meta/config.yml [hunk 4]
```diff
@@ -6,3 +6,8 @@
     - system
     - volume
   hosts: ["localhost:10255"]
+  #ssl.certificate_authorities: ["/etc/pki/root/ca.pem"]
+  #ssl.certificate: "/etc/pki/client/cert.pem"
+  #ssl.key: "/etc/pki/client/cert.key"
```
Comment (jsoriano) [inline metricbeat/module/kubernetes/_meta/config.yml]: ssl options were not introduced on #7134, but I guess that it is something related.

**Pull request merged**